File: leetcode/1869.go
package leetcode

func checkZeroOnes(s string) bool {
	len1, len0, max1, max0 := 0, 0, 0, 0
	for i := range s {
		if s[i] == '0' {
			len0++
			len1 = 0
			if max0 < len0 {
				max0 = len0
			}
		} else {
			len1++
			len0 = 0
			if max1 < len1 {
				max1 = len1
			}
		}
	}
	return max1 > max0
}

File: leetcode/1869.cpp
class Solution {
public:
    bool checkZeroOnes(string s) {
        int len1 = 0, len0 = 0;
        int max1 = 0, max0 = 0;
        for (auto& each : s) {
            if (each == '0') {
                ++len0;
                max0 = max(max0, len0);
                len1 = 0;
            } else {
                ++len1;
                max1 = max(max1, len1);
                len0 = 0;
            }
        }
        return max1 > max0;
    }
};